### ISSUE TITLE
Sound Event Detection 同一随机数种子运行结果不同

### ISSUE THREAD

**Issue by YBRua** [chenxie95/deeplearning_course_sjtu#16]
我注意到SED任务固定了 `torch` 和 `numpy` 的随机数种子，但是今天用同一套Config运行了两次训练和测试，两次的实验结果并不一致（每一个epoch之后的评价指标，以及最后的F1-Score都不一样）。

我这边目前对代码的修改并没有修改初始化随机数种子的部分，也不涉及到 `torch` 或者 `numpy` 之外的随机函数，也谷歌了一下尝试把 `torch.cuda.manual_seed()` 也设置为固定的种子，甚至尝试过revert今天修改的代码，但是都没有解决问题。

我不清楚是我自己这边修改代码的时候翻车了还是这是一个特性，不知道有没有其他同学遇到过类似的情况。

**Comment by zzh-SJTU** [comment]
我也遇到了相同的问题，upsample 函数用的nearest模式，但是还是无法完全复现，不知道是什么问题。

**Comment by wsntxxn** [comment]
[Pytorch 官方](https://pytorch.org/docs/stable/generated/torch.nn.functional.upsample.html) 说该操作不可复现: "This operation may produce nondeterministic gradients when given tensors on a CUDA device."

**Comment by YBRua** [comment]
不过我自己没有主动用到 upsample 或者 interpolate，反而好像是 GRU 加了 Dropout 之后才会无法复现（
